pipeline_tools/get_analysis_metadata.py

```diff
@@ -18,19 +19,81 @@ def get_workflow_id(analysis_output_path):
     url = analysis_output_path
     calls = url.split('/call-')
     workflow_id = calls[1].split('/')[-1]
+    print('Got analysis workflow UUID: {0}'.format(workflow_id))
     with open('workflow_id.txt', 'w') as f:
         f.write(workflow_id)
     return workflow_id
 
 
+def get_adapter_workflow_id(analysis_output_path):
+    """Parse the adapter workflow id from one of its analysis workflow output paths.
+
+    Args:
+        analysis_output_path (str): Path to workflow output file.
+
+    Returns:
+        workflow_id (str): String giving Cromwell UUID of the adapter workflow.
+    """
+    url = analysis_output_path
+    calls = url.split('/call-')
+    adapter_workflow_id = calls[0].split('/')[-1]
+    print('Got adapter workflow UUID: {0}'.format(adapter_workflow_id))
+    with open('adapter_workflow_id.txt', 'w') as f:
+        f.write(adapter_workflow_id)
+    return adapter_workflow_id
+
+
+def get_adapter_workflow_version(cromwell_url,
+                                 adapter_workflow_id,
+                                 http_requests,
+                                 use_caas=False,
+                                 caas_key_file=None):
+    """Get the version of the adapter workflow from its workflow id and write the version to a file so that it is
+    available outside of the get_analysis task.
+
+    Args:
+        cromwell_url (str): Url to the cromwell environment the workflow was run in.
+        adapter_workflow_id (str): String giving Cromwell UUID of the adapter workflow.
+        http_requests: `http_requests.HttpRequests` instance, a wrapper around requests provides better retry and
+                       logging.
+        use_caas (bool): whether or not to use Cromwell-as-a-Service.
+        caas_key_file (str): path to CaaS service account JSON key file.
+
+    Raises:
+        requests.HTTPError: for 4xx errors or 5xx errors beyond the timeout
+    """
+
+    def log_before(workflowId):
+        print('Getting the version for adapter workflow {}'.format(workflowId))
+
+    cromwell_url = cromwell_url
+
+    if use_caas:
+        json_credentials = caas_key_file or "/cromwell-metadata/caas_key.json"
```

Comment:
@rexwangcc can you update this to use the right caas-key depending on the cromwell url? E.g. caas-prod should use the caas-prod key. Right now, there's only one "caas_key" in the docker image that is for caas dev so running pipelines through caas-prod will fail 😞

Reply:
Once we start to use caas-prod, we won't use `caas-dev` anymore, so I think we should only have 1 caas_key.json instead of differentiating caas-prod and caas-dev in pipeline-tools 's code base, which will require us to parse the url and check prod/dev in the parsed strings. Do you mind I leave this (updating the service account in cromwell-metadata docker image) as a future work, along with other works that migrating to caas-prod? If so, the only thing we need to do there is rebuild the image and push it to gcr, no need to change anything in pipeline-tools

Reply:
Sure, that makes sense to me